### PR TITLE
Use `button` CSS vars for border-color to support a feature flag

### PR DESCRIPTION
**Pull request by langermank**
This has no visual change, and is for a test in dotcom where I'm targeting these CSS variables. 

### Changelog

- Use Primitives v8 color tokens for the `default` and `outline` border colors including a fallback

### Rollout strategy

- [ ] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan

### Testing & Reviewing

There should be **no** visual changes in this PR. Check that the visual regression snapshots look normal.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: .storybook/primitives-v8.css [hunk 1]
```diff
@@ -10,12 +10,14 @@
...
-@import '@primer/primitives/tokens-next-private/css/functional/themes/light.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/light-tritanopia.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/light-high-contrast.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/light-colorblind.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/dark.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/dark-colorblind.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/dark-dimmed.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/dark-high-contrast.css';
-@import '@primer/primitives/tokens-next-private/css/functional/themes/dark-tritanopia.css';
+/* temporarily disabiling these to ensure visual regression tests work on production code
```
Comment (siddharthkp) [inline .storybook/primitives-v8.css:13]: do we want to keep these disabled in storybook?

File: src/Button/styles.ts [hunk 2]
```diff
@@ -9,11 +9,11 @@ export const getVariantStyles = (variant: VariantType = 'default', theme?: Theme
       boxShadow: `${theme?.shadows.btn.shadow}, ${theme?.shadows.btn.insetShadow}`,
       '&:hover:not([disabled])': {
         backgroundColor: 'btn.hoverBg',
-        borderColor: 'btn.hoverBorder',
+        borderColor: `var(--button-default-borderColor-hover, ${theme?.colors.btn.hoverBorder})`,
```
Comment (siddharthkp) [inline src/Button/styles.ts:12]: has a fallback 10/10

File: src/__tests__/__snapshots__/ActionMenu.test.tsx.snap [hunk 3]
```diff
@@ -25,7 +25,7 @@ exports[`ActionMenu renders consistently 1`] = `
 .c1 {
   border-radius: 6px;
   border: 1px solid;
-  border-color: rgba(31,35,40,0.15);
+  border-color: var(--button-default-borderColor-rest,rgba(31,35,40,0.15));
```
Comment (siddharthkp) [inline src/__tests__/__snapshots__/ActionMenu.test.tsx.snap:28]: ✅  makes sense, ActionMenu uses Button

**Comment by siddharthkp** [comment]
just checking, are the visual changes to TextInput (and components that use TextInput) intentional?

visual diff for textinput, the border is darker now

File: .storybook/primitives-v8.css (hunk 1 above)
Comment (langermank) [reply .storybook/primitives-v8.css:13]: I'd like to add a toolbar addon to let you toggle them on, like how we have it in Primer View Components Lookbook

primitives toggle

Comment (siddharthkp) [reply .storybook/primitives-v8.css:13]: Good idea, should we do that in a different PR though and get this one quickly?

Comment (langermank) [reply .storybook/primitives-v8.css:13]: Yeah totally agree

**Review by siddharthkp — APPROVED** [review]
LGTM, the shapshot changes are intentional!

**Pull request merged**